src/nn/prepo.rs

```diff
@@ -14,99 +19,109 @@ extern "C" {
     ) -> root::Result;
-}
-extern "C" {
+
     #[link_name = "\u{1}_ZN2nn5prepo10PlayReport3AddEPKcS3_"]
     pub fn PlayReport_Add2(
         this: *mut PlayReport,
-        arg1: *const u8,
-        arg2: *const u8,
+        key: *const u8,
+        value: *const u8,
     ) -> root::Result;
-}
-extern "C" {
+
+    #[link_name = "\u{1}_ZN2nn5prepo10PlayReport3AddEPKcRKNS0_10Any64BitIdE"]
+    pub fn PlayReport_Add3(
+            this: *mut PlayReport,
+            key: *const u8,
+            value: Any64BitId,
+    ) -> root::Result;
+
     #[link_name = "\u{1}_ZN2nn5prepo10PlayReport4SaveEv"]
     pub fn PlayReport_Save(this: *mut PlayReport) -> root::Result;
-}
-extern "C" {
+
     #[link_name = "\u{1}_ZN2nn5prepo10PlayReport4SaveERKNS_7account3UidE"]
     pub fn PlayReport_Save1(
         this: *mut PlayReport,
-        arg1: *const root::nn::account::Uid,
+        uid: *const root::nn::account::Uid,
     ) -> root::Result;
-}
-extern "C" {
+
     #[link_name = "\u{1}_ZN2nn5prepo10PlayReportC1Ev"]
     pub fn PlayReport_PlayReport(this: *mut PlayReport);
 }
 impl PlayReport {
     #[inline]
-    pub unsafe fn SetEventId(&mut self, arg1: *const u8) -> root::Result {
-        PlayReport_SetEventId(self, arg1)
+    pub fn SetEventId(&mut self, arg1: *const u8) -> root::Result {
+        unsafe { PlayReport_SetEventId(self, arg1) }
     }
     #[inline]
-    pub unsafe fn SetBuffer(&mut self) -> root::Result {
-        PlayReport_SetBuffer(self)
+    pub fn SetBuffer(&mut self) -> root::Result {
+        unsafe { PlayReport_SetBuffer(self) }
     }
     #[inline]
-    pub unsafe fn Add(
+    pub fn AddLong(
         &mut self,
-        arg1: *const u8,
-        arg2: i64,
+        key: *const u8,
+        value: i64,
     ) -> root::Result {
-        PlayReport_Add(self, arg1, arg2)
+        unsafe { PlayReport_Add(self, key, value) }
     }
     #[inline]
-    pub unsafe fn Add1(
+    pub fn AddDouble(
         &mut self,
-        arg1: *const u8,
-        arg2: f64,
+        key: *const u8,
+        value: f64,
     ) -> root::Result {
-        PlayReport_Add1(self, arg1, arg2)
+        unsafe { PlayReport_Add1(self, key, value) }
     }
     #[inline]
-    pub unsafe fn Add2(
+    pub fn AddString(
         &mut self,
-        arg1: *const u8,
-        arg2: *const u8,
+        key: *const u8,
+        value: *const u8,
+    ) -> root::Result {
+        unsafe { PlayReport_Add2(self, key, value) }
+    }
+    #[inline]
+    pub fn AddAny64BitID(
+        &mut self,
+        key: *const u8,
+        value: Any64BitId,
     ) -> root::Result {
-        PlayReport_Add2(self, arg1, arg2)
+        unsafe { PlayReport_Add3(self, key, value) }
     }
     #[inline]
-    pub unsafe fn Save(&mut self) -> root::Result {
-        PlayReport_Save(self)
+    pub fn Save(&mut self) -> root::Result {
+        unsafe { PlayReport_Save(self) }
     }
     #[inline]
-    pub unsafe fn Save1(
+    pub fn SaveWithAccount(
         &mut self,
```

Comment:
remove the &mut part of it, make it just "self", that way the struct is consumed after the caller uses Save and can't use the struct anymore